sql/sql_yacc.yy

```diff
@@ -2908,15 +2918,15 @@ create_index_stmt:
                                              $12.algo.get_or_default(),
                                              $12.lock.get_or_default());
           }
-        | CREATE FULLTEXT_SYM INDEX_SYM ident ON_SYM table_ident
+        | create_opt FULLTEXT_SYM INDEX_SYM ident ON_SYM table_ident
           '(' key_list_with_expression ')' opt_fulltext_index_options opt_index_lock_and_algorithm
           {
             $$= NEW_PTN PT_create_index_stmt(YYMEM_ROOT, KEYTYPE_FULLTEXT, $4,
                                              NULL, $6, $8, $10,
                                              $11.algo.get_or_default(),
                                              $11.lock.get_or_default());
           }
-        | CREATE SPATIAL_SYM INDEX_SYM ident ON_SYM table_ident
+        | create_opt SPATIAL_SYM INDEX_SYM ident ON_SYM table_ident
```

Comment:
I'm seeing bison register another conflict:

/home/herman/rocks-mysql/8.0/sql/sql_yacc.yy: error: shift/reduce conflicts: 110 found, 109 expected

The conflict comes from the above rule and rule for CREATE SPATIAL_SYM REFERENCE_SYM". Changing "CREATE SPATIAL_SYM REFERENCE_SYM" to "create_opt SPATIAL_SYM REFERENCE_SYM" for the create_srs_stmt production resolves this, but also changing the create_srs_stmt -> CREATE OR_SYM REPLACE_SYM SPATIAL_SYM REFERENCE_SYM causes another conflict.

What is the best way to resolve this, if any?

Reply:
I would approach this from a different angle:

* do you want to add the high priority option to other commands?
* if yes, do you actually want those to be logical? (e.g. if one variant gets it, others get it too)

I'm not sure if it can be correctly resolved if one of the answers is no. (and the conflict was there in 5.6 too)

Reply:
Yes. We should add high priority ddl to those statements too and having them be logical would be fine.

Reply:
To clarify: this statement is the `CREATE SPATIAL REFERENCE SYSTEM`, not the `CREATE SPATIAL INDEX`, and I don't think high priority ddl will do anything for it?

But if I do add the high priority option to it, the warning is fixable, so I updated my PR with that.

If I do not add the option to it, the conflict is only resolvable with a larger refactoring, which I don't think we should do.

Reply:
Yes, adding high priority DDL to those statements to fix the warning without a major refactoring is fine. That's the temporary fix I'm using at the moment to remove the warning. Since high priority ddl is basically a noop for it, I think it would be okay.